modules/33acrossBidAdapter.js

```diff
@@ -194,18 +194,44 @@ function _createServerRequest({bidRequest, gdprConsent = {}, uspConsent, pageUrl
   // therefore in ad targetting process
   ttxRequest.id = bidRequest.bidId;
 
-  // Set GDPR related fields
-  ttxRequest.user = {
-    ext: {
-      consent: gdprConsent.consentString
-    }
-  };
-  ttxRequest.regs = {
-    ext: {
-      gdpr: (gdprConsent.gdprApplies === true) ? 1 : 0,
-      us_privacy: uspConsent || null
-    }
-  };
+  if (gdprConsent.consentString) {
+    ttxRequest.user = setExtension(
+      ttxRequest.user,
+      'consent',
+      gdprConsent.consentString
+    )
+  }
+
+  if (Array.isArray(bidRequest.userIdAsEids)) {
+    ttxRequest.user = setExtension(
+      ttxRequest.user,
+      'eids',
+      bidRequest.userIdAsEids
+    )
+  }
+
+  if (gdprConsent.gdprApplies === true) {
+    ttxRequest.regs = setExtension(
+      ttxRequest.regs,
+      'gdpr',
+      1
+    )
```

Comment:
wondering if we can avoid the branching for this particular case.

e.g.
```
ttxRequest.regs = setExtension({
  ttxRequest.regs,
  'gdpr',
  Number(gdprConsent.gdprApplies)
})
```

Reply:
Can you elaborate more on why you are recommending avoiding branching given that the branching makes the logic more explicit? Performance?

Reply:
Before I weight on non using conditional branching, I don’t want you to think that I am forcing this feedback change, I forgot to mention that's optional, your code looks good and more important, the behavior is correct.

I like to avoid conditional branching in order to keep the code more concise, less verbose and DRY. On the other hand there are some ongoing discussions in the JS community about conditional branching, if it should be considered a code smell or not, and the alternatives for some cases (like polymorphism):

https://levelup.gitconnected.com/conditional-statements-are-a-code-smell-here-is-your-relief-38e50c023708
https://stackoverflow.com/questions/1554180/why-is-the-if-statement-considered-evil

In the SIC team we decided to avoid branching in order to implement behavior:
https://docs.google.com/document/d/1vji7ECS51SCehm3P3sTugrJIOlnpDLTrKpSLn1ND__0/edit?usp=sharing

but I understand that the prebid adapter guidelines could be different. I saw an opportunity to remove a few lines of code but feel free to apply the change or not.

Reply:
I have adopted this suggestion as it makes sense. We are after all setting a value of 1 or 0.